### PR TITLE
Add parent lookup descriptor

### PR DESCRIPTION
**Pull request by ClaasRostock**
### Added
* Added a new generic descriptor API, `ParentLookup`, as an optional alternative to implementing child-side `find_parent` methods.

## Summary by Sourcery

Introduce a generic ParentLookup descriptor as an optional, typed API for resolving parent instances from child objects and update project metadata for a new patch release.

New Features:
- Add a generic ParentLookup descriptor that delegates to the lookup registry to resolve a child instance’s parent at attribute access time.
- Expose the ParentLookup descriptor from the parent_lookup package public API.

Enhancements:
- Document descriptor-based parent lookup as an alternative to explicit find_parent methods in the README.
- Add a concrete implementation plan and chat log artifacts for the ParentLookup descriptor feature under the plans directory.

Build:
- Bump project version and Sphinx docs release to 0.2.3 and update changelog links for the new release.

Documentation:
- Update README with an example demonstrating descriptor-based parent lookup.

Tests:
- Add descriptor-based parent/child fixture modules and a dedicated test suite to validate ParentLookup behavior, including future-annotations scenarios and deepcopy handling.

### CODE REVIEW

File: tests/lookup/test_parent_lookup_descriptor.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+def test_descriptor_finds_parent() -> None:
+    # Prepare
+    parent = DescriptorParent()
+    child = DescriptorChild()
+    parent.add_child(child)
+    # Execute
+    found_parent = child.parent
+    # Assert
+    assert found_parent is parent
```
[the remark this answers is not in the document]
Comment (ClaasRostock) [reply tests/lookup/test_parent_lookup_descriptor.py:21]: @copilot:
Add a test that verifies the descriptor returns None when no parent is registered for a child instance.

All existing tests assume a parent is registered in `lookup_registry`. Please also add a case (e.g. `test_descriptor_returns_none_when_no_parent_registered`) that creates a `DescriptorChild` without registering it under any parent and asserts `child.parent` is `None`. This will lock in the expected fallback behavior and help prevent regressions in the registry/descriptor interaction.
